core/src/main/java/org/frankframework/util/flow/graphviz/GraphvizEngine.java

```diff
@@ -60,25 +65,42 @@ public GraphvizEngine() throws IOException {
 
 	/**
 	 * Create a new GraphvizEngine instance
-	 * @param graphvizVersion version of the the VisJs engine to initiate
+	 * @param graphvizVersion version of the VisJs engine to initiate
 	 * @throws IOException
 	 */
 	public GraphvizEngine(String graphvizVersion) throws IOException {
-		if(StringUtils.isNotEmpty(graphvizVersion)) {
+		if (StringUtils.isNotEmpty(graphvizVersion)) {
 			this.graphvizVersion = graphvizVersion;
 		}
 
 		try {
-			Format format = Format.valueOf(fileFormat.toUpperCase());
+			Format format = Format.valueOf(FILE_FORMAT);
 			defaultOptions = Options.create().format(format);
-			if(log.isDebugEnabled()) log.debug("Setting Graphviz options to ["+defaultOptions+"]");
-		}
-		catch(IllegalArgumentException e) {
-			throw new IllegalArgumentException("unknown format["+fileFormat.toUpperCase()+"], must be one of "+Format.values());
+			if (log.isDebugEnabled()) log.debug("Setting Graphviz options to [" + defaultOptions + "]");
+		} catch (IllegalArgumentException e) {
+			throw new IllegalArgumentException("unknown format[" + FILE_FORMAT + "], must be one of " + Format.values());
 		}
 
 		//Create the GraphvizEngine, make sure it can find and load the required libraries
 		getEngine();
+
+		CleanupEngineAction cleanupEngineAction = new CleanupEngineAction(engine);
+		cleanable = cleaner.register(this, cleanupEngineAction);
+	}
+
+	private static class CleanupEngineAction implements Runnable {
```

Comment:
Deze code zie ik nu 3x voorbijkomen, is het niet netter hier een 'generic' van te maken? 
`CleanupAction<T extends AutoCloseable> implements runnable`

Reply:
> Deze code zie ik nu 3x voorbijkomen, is het niet netter hier een 'generic' van te maken? `CleanupAction<T extends AutoCloseable> implements runnable`

Maar de `run()` method moet iedere keer iets anders doen specifiek voor de class waar deze bij hoort. En het moet ook een `static` class zijn, die geen references mag hebben naar de class-instance waarvoor je de clean-actie wil doen. (Wel naar members van die instance, maar niet naar die instance zelf, anders blijven er references aanwezig naar de instance en heb je een echt memory-leak).

Reply:
Eens met Tim. Als het verder goed is, laten we het dan mergen.